internal/config/flag.go

```diff
@@ -0,0 +1,76 @@
+package config
+
+import (
+	"errors"
+	"flag"
+	"log"
+
+	"github.com/caarlos0/env/v11"
+)
+
+type Config struct {
+	ServerHostPort string `env:"ADDRESS"`
+	ReportInterval int    `env:"REPORT_INTERVAL"`
+	PollInterval   int    `env:"POLL_INTERVAL"`
+}
+
+var appConfig *Config
+
+func GetConfig() *Config {
+	if appConfig == nil {
+		c, err := initAppConfig()
+		if err != nil {
+			log.Fatal(err)
+		}
+		appConfig = c
+	}
+	return appConfig
+}
+
+const defaultReportIntervalSeconds = 10
+const defaultPollIntervalSeconds = 2
+
+func initAppConfig() (*Config, error) {
+	var (
+		sh string
+		ri int
+		pi int
+	)
+
+	c := Config{}
+
+	err := env.Parse(&c)
+	if err != nil {
+		return nil, errors.New(err.Error())
```

Comment:
```suggestion
		return nil, err		
```
Это не оборачивание ошибки получается, а пустая операция по разворачиванию в строку и заворачиванию обратно в error. При этом оригинальный тип ошибки теряется, так что данный метод является антипаттерном.